src/main.ts

```diff
@@ -975,7 +983,19 @@ function generateService(
     maybeAddComment(info, text => (requestFn = requestFn.addJavadoc(text)));
 
     requestFn = requestFn.addParameter('request', requestType(typeMap, methodDesc));
-    requestFn = requestFn.returns(responsePromise(typeMap, methodDesc));
+
+    // Use metadata as last argument for interface only configuration
+    if (!options.outputClientImpl && !options.outputEncodeMethods && !options.outputJsonMethods && options.addGrpcMetadata) {
+      requestFn = requestFn.addParameter('metadata?', "Metadata@grpc");
+    }
+
+    // Return observable for interface only configuration and passing returnObservable=true
+    if (!options.outputClientImpl && !options.outputEncodeMethods && !options.outputJsonMethods && options.returnObservable) {
```

Comment:
Simplified with same rationale, but in retrospect caveat about outputclientimpl, as above.